subscription-manager.spec

```diff
@@ -681,6 +681,8 @@ find %{buildroot} -name \*.py* -exec touch -r %{SOURCE0} '{}' \;
     %tmpfiles_create %{_tmpfilesdir}/subscription-manager.conf
 %else
     %systemd_post rhsmcertd.service
+    %systemd_post rhsm.service
+    systemctl restart rhsm.service
```

Comment:
Reference of the RPM macros: https://github.com/systemd/systemd/blob/46f881e32127cce03118dcd0471ded12d08d50b2/src/rpm/macros.systemd.in

`%systemd_post` in practice only installs the new service, and it does not do anything more than that

What seems to be needed here is `%systemd_postun_with_restart`, which seems to be marking the unit as "needs to be restarted", and then the rpm file triggers will make systemd restart the units marked as "no be restarted". Please note that the right place for that is in `%postun`, not `%post`.